### PR TITLE
JS: Improve insufficient pw hash query

### PR DESCRIPTION
**Pull request by smiddy007**
Added to the crypto-js library module to match progressive hashing in order to improve InsufficientPasswordHash.ql.  Also added example js files and renamed existing example files as being Node-js examples.

### CODE REVIEW

**Comment by smiddy007** [comment]
edit

**Review by asgerf — COMMENTED** [review]
Thanks for the contribution! Just one comment so far

File: javascript/ql/lib/semmle/javascript/frameworks/CryptoLibraries.qll [hunk 1]
```diff
@@ -325,13 +367,40 @@ private module CryptoJS {
...
   private class Apply extends CryptographicOperation::Range instanceof API::CallNode {
     API::Node input;
     CryptographicAlgorithm algorithm; // non-functional
+    InstantiatedAlgorithm instantiation;
 
     Apply() {
       this = getEncryptionApplication(input, algorithm) or
-      this = getDirectApplication(input, algorithm)
+      this = getDirectApplication(input, algorithm) or
+      this = getUpdatedApplication(input, instantiation)
```
[the remark this answers is not in the document]
Comment (smiddy007) [reply javascript/ql/lib/semmle/javascript/frameworks/CryptoLibraries.qll]: Thank you for the help!

**Review by asgerf — APPROVED** [review]
LGTM, thanks for the contribution!

**Pull request merged**